src/evaluate/evaluator/question_answering.py

```diff
@@ -0,0 +1,255 @@
+                the name of the column cointaing the identification field of the question and answer pair in the
+                dataset specified by `data`.
+            label_column (`str`, defaults to `"answers"`):
+                the name of the column containing the answers in the dataset specified by `data`.
+        Return:
+            A `Dict`. The keys represent metric keys calculated for the `metric` spefied in function arguments. For the
+            `"simple"` strategy, the value is the metric score. For the `"bootstrap"` strategy, the value is a `Dict`
+            containing the score, the confidence interval and the standard error calculated for each metric key.
+
+        Examples:
+        ```python
+        >>> from evaluate import evaluator
+        >>> from datasets import Dataset, load_dataset
+        >>> e = evaluator("question-answering")
+        >>> data = load_dataset("squad", split="validation[:2]")
+        >>> results = e.compute(
+        >>>     model_or_pipeline="sshleifer/tiny-distilbert-base-cased-distilled-squad",
+        >>>     data=data,
+        >>>     metric="squad",
+        >>> )
+        ```
+
+        <Tip>
+
+        Datasets where the answer may be missing in the context are supported, for example SQuAD v2 dataset. If using transformers pipeline
+        with models trained on this type of data, make sure to pass `handle_impossible_answer=True` as an argument to the pipeline.
+
+        </Tip>
+
+        ```python
+        >>> from evaluate import evaluator
+        >>> from datasets import Dataset, load_dataset
+        >>> from transformers import pipeline
+        >>> task_evaluator = evaluator("question-answering")
+        >>> data = load_dataset("squad_v2", split="validation[:2]")
+        >>> pipe = pipeline(
+        >>>     task="question-answering",
+        >>>     model="sshleifer/mrm8488/bert-tiny-finetuned-squadv2",
+        >>>     handle_impossible_answer=True
+        >>> )
+        >>> results = task_evaluator.compute(
+        >>>     model_or_pipeline=pipe,
+        >>>     data=data,
+        >>>     metric="squad_v2",
+        >>> )
+        ```
+        """
+        result = {}
+
+        metric_inputs, pipe_inputs = self.prepare_data(
+            data=data,
+            question_column=question_column,
+            context_column=context_column,
+            id_column=id_column,
+            label_column=label_column,
+        )
+
+        squad_v2_schema = self.is_squad_v2_schema(data=data, label_column=label_column)
+
+        if squad_v2_schema and metric.name == "squad":
+            logger.warn(
+                "The dataset has SQuAD v2 format but you are using the SQuAD metric. Consider passing the 'squad_v2' metric."
+            )
+        if not squad_v2_schema and metric.name == "squad_v2":
+            logger.warn(
+                "The dataset has SQuAD v1 format but you are using the SQuAD v2 metric. Consider passing the 'squad' metric."
+            )
```

Comment:
We have to do this after `self.prepare_metric` otherwise `metric` could just be a string.